sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4520,9 +4520,12 @@ def _execute_task_one_node(self, handle: CloudVmRayResourceHandle,
 
         command_for_node = task.run if isinstance(task.run, str) else None
         use_sudo = isinstance(handle.launched_resources.cloud, clouds.Local)
+        envs_with_external = copy.deepcopy(task.envs)
+        envs_with_external['SKYPILOT_EXTERNAL_IPS'] = '\n'.join(
```

Comment:
This will include the external ips for all the nodes, but the task may only be run on part of the nodes in the cluster. We probably need to change the `stable_cluster_internal_ips` in L4502 and filter the external IPs for the nodes the task is scheduled on. Please refer to the place we get the internal IPs: https://github.com/skypilot-org/skypilot/blob/94b4ce31fb44f9232bce93b1febdf3c1d1291bca/sky/backends/cloud_vm_ray_backend.py#L506